src/v/datalake/tests/test_data_writer.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "datalake/schemaless_translator.h"
+#include "iceberg/datatypes.h"
+
+#include <cstdint>
+namespace datalake {
+
+class test_data_writer_factory {
+public:
+    class data_writer_result {
+    public:
+        data_writer_result() = default;
+        int64_t row_count{};
+    };
+    class data_writer {
+    public:
+        explicit data_writer(iceberg::struct_type&& schema)
+          : _schema(std::move(schema))
+          , _result() {}
+
+        // TODO: error type?
+        bool add_data_struct(
+          schemaless_event_data& /* data */, int64_t /* approx_size */) {
```

Comment:
I was anticipating that batching will probably have both a row limit and size limit for row groups.

Reply:
But isn't this something that the underlying batcher would know as it's constructing the row groups? Wouldn't the batching be done by the size in memory of the in-progress rowgroup rather than the size of input data?